staging/mausezahn.c

```diff
@@ -671,6 +671,15 @@ int getopts (int argc, char *argv[])
 				tx.ip_src_rand = 1;
 				tx.ip_src_h  = (u_int32_t) ( ((float) rand()/RAND_MAX)*0xE0000000); //this is 224.0.0.0
 			}
+			else if (ipv6_mode && (strstr(tx.ip_src_txt, ".") || strstr(tx.ip_dst_txt, "."))) {
+			       fprintf(stderr, "You are using IPv6 mode but are providing IPv4 src/dst addresses.\n");
+			       return 1;
+			}
+			else if (!ipv6_mode && (strstr(tx.ip_src_txt, ":") || strstr(tx.ip_dst_txt, ":"))) {
+			       fprintf(stderr, "You are using IPv4 mode (the default) but are providing IPv6 src/dst addresses.\n");
+			       fprintf(stderr, "Please use the -6 option for IPv6 mode.\n");
+			       return 1;
+			}
```

Comment:
Thanks for spotting that issue! To address 1. immediately, I have opened #167, since it's rather serious and could cause problems if anyone is building from master. (Also that check wasn't as necessary in the first place).

I'll work on moving the check to a different section and using AF_INET.